config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1183,11 +1190,15 @@ protected void configure(ServerHttpSecurity http) {
 					MediaType.TEXT_HTML);
 			htmlMatcher.setIgnoredMediaTypes(Collections.singleton(MediaType.ALL));
 			Map<String, String> urlToText = http.oauth2Login.getLinks();
+			String authenticationEntryPointRedirectPath;
 			if (urlToText.size() == 1) {
-				http.defaultEntryPoints.add(new DelegateEntry(htmlMatcher, new RedirectServerAuthenticationEntryPoint(urlToText.keySet().iterator().next())));
+				authenticationEntryPointRedirectPath = urlToText.keySet().iterator().next();
 			} else {
-				http.defaultEntryPoints.add(new DelegateEntry(htmlMatcher, new RedirectServerAuthenticationEntryPoint("/login")));
+				authenticationEntryPointRedirectPath = "/login";
 			}
+			RedirectServerAuthenticationEntryPoint entryPoint = new RedirectServerAuthenticationEntryPoint(authenticationEntryPointRedirectPath);
+			entryPoint.setRequestCache(http.requestCache.requestCache);
```

Comment:
I'm not sure if `RedirectServerAuthenticationEntryPoint` needs the same `requestCache`? I think we only want to save the request when an authentication fails only. Does a test fail without this change?

Reply:
Hey @jgrandja. That's my main, issue. I'm using a react application on a different port and and want to save the "referer" header into a `RequestCache` instead of the normal entry point. I'm unable to get to the `RequestCache` on the `RedirectServerAuthenticationEntryPoint` without this change. I'm thinking something similar to FormLoginSpec: https://github.com/spring-projects/spring-security/blob/master/config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java#L3118

Reply:
I've found a better way to accomplish what I'm looking for by doing employing x-forward* on my proxy and `server.forward-headers-strategy: framework` on my spring-cloud-gateway (see proxy [reference](https://docs.spring.io/spring-security/site/docs/current/reference/html/webflux-oauth2.html#webflux-oauth2-login-sample-redirect)) but still wonder...

There are at least 3 places in the oauth2login where a `RequestCache` is set (defaultEntryPoint, authenticationSuccessHandler, and OAuth2AuthorizationRequestRedirectWebFilter) . Currently, only `OAuth2AuthorizationRequestRedirectWebFilter` is configurable using the `RequestCacheSpec`. Should they all have their own independent `RequestCache` implementations and how should oauth2Login differ from formLogin WRT `RequestCache`?

Reply:
@jgrandja I am under the impression that all components that participate in a flow would need to share a request cache. And the cache would need to remain the same between components that save the request and read the saved request.

Reply:
@fhanik @sdoxsee Sorry for the confusion on my end. I made the assumption that a failed request is saved in `ExceptionTranslationWebFilter` instead of `RedirectServerAuthenticationEntryPoint`, as this is the way the Servlet counterpart `ExceptionTranslationFilter` is implemented. However, the reactive implementation is different and indeed the `RedirectServerAuthenticationEntryPoint` is responsible for saving the failed request.

@sdoxsee 

> Should they all have their own independent RequestCache implementations

No, all components participating in the same flow should share the same request cache.